codecs/src/aper/decode/decode_charstrings.rs

```diff
@@ -133,31 +54,41 @@ pub fn decode_utf8_string(
         ub,
         is_extensible
     );
+    decode_string(data, lb, ub, is_extensible)
+}
 
-    let (_val_lower, _val_higher) = (0u8, 255u8);
-
-    let num_bits = 8; // N = 74, B = 7, B2 = 8
-
-    let length = decode_length_determinent(data, None, None, false)?;
-    let mut out = String::new();
-    if length > 0 {
-        let length = length * num_bits;
-
-        if length > 16 {
-            data.decode_align()?;
-        }
+fn decode_string(
```

Comment:
Good that you got rid of the duplicate code. I don't exactly remember now why these functions had a duplicated code, may be something that was like - I will fix later.